### PR TITLE
Add "Golden copy" test runner with single test

### PR DESCRIPTION
**Pull request by mr0re1**
### Submission Checklist

* [ ] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [ ] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [ ] Is unit test coverage still above 80%?
* [ ] Have you updated all applicable documentation?
* [ ] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: tools/validate_configs/test_configs/igc_pkr_test.yaml
Comment (tpdownes) [inline tools/validate_configs/test_configs/igc_pkr_test.yaml:44]: "hellow" is a typo in my original example from #1193. Change it to "hello"

File: .pre-commit-config.yaml
Comment (tpdownes) [inline .pre-commit-config.yaml:1]: Would a [global exclude](https://github.com/pre-commit/pre-commit/pull/617) be helpful here?

File: tools/validate_configs/golden_copies/packer_igc/zero/outputs.tf
Comment (tpdownes) [inline tools/validate_configs/golden_copies/packer_igc/zero/outputs.tf:1]: #1199 is going to change white space here (eliminates trailing newline and adds back newline between license and first block).

A lot of these files have an extra newline at end I've been slowly working through. Might make sense to do that after this PR as a test of the change. I will plan to merge #1199 after this.

(no action needed)

File: tools/validate_configs/validate_golden_copy.sh
Comment (tpdownes) [inline tools/validate_configs/validate_golden_copy.sh:56]: it might make sense to TRAP the cleanup. e.g. (first link I found). do you agree?

https://www.linuxjournal.com/content/use-bash-trap-statement-cleanup-temporary-files

**Review by tpdownes — APPROVED** [review]
Please fix typo before merging. Please give consideration to the TRAP recommendation for the validation script, though it does not block merging if you don't see the value.

File: .pre-commit-config.yaml
Comment (mr0re1) [reply .pre-commit-config.yaml:1]: done

File: tools/validate_configs/validate_golden_copy.sh
Comment (mr0re1) [reply tools/validate_configs/validate_golden_copy.sh:56]: I would prefer to consider it next time. This script is copy-paste of validate_config, we can change them both at once later.

**Review by tpdownes — APPROVED** [review]

**Pull request merged**
